Increase the accuracy of Flash-based seeking

dmlap opened the pull request:
Keep track of a mapping between PTS positions and the media timeline so we can determine a target PTS value when seeking occurs. Ensure that resetting the timestampOffset (that is, crossing a discontinuity) causes the mapping to get refreshed. Decode the arguments being passed into the mock SWF in testing so the error messages can be more helpful when tests fail.

dmlap committed:
Increase the accuracy of Flash-based seeking
Keep track of a mapping between PTS positions and the media timeline so we can determine a target PTS value when seeking occurs. Ensure that resetting the timestampOffset (that is, crossing a discontinuity) causes the mapping to get refreshed. Decode the arguments being passed into the mock SWF in testing so the error messages can be more helpful when tests fail.

imbcmdth commented:
LGTM but I am going to wait until you have pulled in and tested the latest changes to HLS since this might need some updating.

dmlap committed:
Add a no-op remove for Flash
HTML MSE has a real and usable implementation of remove() that we use to clear buffered content that may have been "misplaced" in the timeline do to discontinuties. Flash always blows away the buffer on seek so it's not necessary for that case. We still don't want an error to be thrown though, so add a no-op version.
